tm2/pkg/bft/rpc/client/batch.go

```diff
@@ -142,16 +142,16 @@ func (b *RPCBatch) ABCIQuery(path string, data []byte) error {
 }
 
 func (b *RPCBatch) ABCIQueryWithOptions(path string, data []byte, opts ABCIQueryOptions) error {
+	params := map[string]any{
+		"path":  path,
+		"data":  data,
+		"prove": opts.Prove,
+	}
+	if opts.Height != 0 {
```

Comment:
Just curious: isn't this misleading? Specifying 0 will cause an error down the line if I'm not mistaken, but in this case it will just continue on

Reply:
hmm.. yes it would be clearer to only add the key if `opts.Height` is greater than 0